ouroboros-network/src/Ouroboros/Network/Chain.hs

```diff
@@ -289,22 +293,23 @@ findFirstPoint
   :: HasHeader block
   => [Point block]
   -> Chain block
-  -> Maybe (Point block)
-findFirstPoint [] _     = Nothing
+  -> Point block
+findFirstPoint [] _     = Origin
```

Comment:
This is where it becomes strange since now `Origin` means two things: not found and if `Origin` was included in the `ps :: [Point block]` list it may return `Origin` because it's found. I'd argue that this is not important, because including `Origin` makes not much sense in the first place.

Reply:
`Origin` is on every `Chain block` so yes, `findFirstPoint (Origin : ps) chain = findFirstPoint ps chain`. It's not so strange. There's a similar story for `False` and `or :: [Bool] -> Bool`.

Reply:
On second thought, it actually is important to distinguish the case when none of the given points are on the chain. Comparison to `or` isn't good, since `findFirstPoint` is primarily about membership in the `[Point block]` list. I'll change it back to give `Maybe (Point block)`

Reply:
This makes sense.